packages/forms/src/model/form_group.ts

```diff
@@ -281,25 +410,29 @@ export class FormGroup extends AbstractControl {
    * is updated. When false, no events are emitted. The configuration options are passed to
    * the {@link AbstractControl#updateValueAndValidity updateValueAndValidity} method.
    */
-  override patchValue(
-      value: {[key: string]: any}, options: {onlySelf?: boolean, emitEvent?: boolean} = {}): void {
+  override patchValue(value: ɵFormGroupValue<TControl>, options: {
+    onlySelf?: boolean,
+    emitEvent?: boolean
+  } = {}): void {
     // Even though the `value` argument type doesn't allow `null` and `undefined` values, the
     // `patchValue` can be called recursively and inner data structures might have these values, so
     // we just ignore such cases when a field containing FormGroup instance receives `null` or
     // `undefined` as a value.
     if (value == null /* both `null` and `undefined` */) return;
-
-    Object.keys(value).forEach(name => {
-      if (this.controls[name]) {
-        this.controls[name].patchValue(value[name], {onlySelf: true, emitEvent: options.emitEvent});
+    (Object.keys(value) as Array<keyof TControl>).forEach(name => {
+      if ((this.controls as any)[name]) {
+        (this.controls as any)[name]!.patchValue(
```

Comment:
Possible refactoring:
```suggestion
      // ... comment on why we use `as any` here ...
      const control = (this.controls as any)[name];
      control?.patchValue(
```

Reply:
Fixed.